### DISCUSSION TITLE
use-package recipe with :vc

### DISCUSSION THREAD

**Discussion by zeeros** [kickingvegas/casual-dired#93] [Q&A]
I'm using `Emacs 30.0.90`. I've used this kind of recipe (`use-package` with `:vc` keyword) also with magit, which has a similar structure, without encountering any error.
When I use the following recipe

```
(use-package casual-dired
  :vc (:url "https://github.com/kickingvegas/casual-dired" :rev "v1.8.1" :lisp-dir "lisp/")
  :bind (:map dired-mode-map
              ("C-o" . #'casual-dired-tmenu)
              ("s" . #'casual-dired-sort-by-tmenu)
              ("/" . #'casual-dired-search-replace-tmenu)))
```

I get the following error in the `*Compile Log*` buffer

`Error (bytecomp): Cannot open load file: No such file or directory, casual-dired-test-utils [6 times]
`

As expected, `use-package` downloads the entire repo

```
casual-dired/
├── casual-dired-autoloads.el
├── casual-dired-autoloads.elc
├── casual-dired-pkg.el
├── docs
│   ├── developer.org
│   └── images
│       ├── casual-dired-screenshot.png
│       ├── casual-dired-screenshot-unicode.png
│       ├── casual-dired-search-replace.png
│       ├── casual-dired-settings-screenshot.png
│       ├── casual-dired-sort-by-screenshot.png
│       ├── casual-dired-utils-screenshot.png
│       └── default-yellow.png
├── LICENSE
├── lisp
│   ├── casual-dired-autoloads.el
│   ├── casual-dired.el
│   ├── casual-dired.elc
│   ├── casual-dired-settings.el
│   ├── casual-dired-settings.elc
│   ├── casual-dired-sort-by.el
│   ├── casual-dired-sort-by.elc
│   ├── casual-dired-utils.el
│   ├── casual-dired-utils.elc
│   ├── casual-dired-variables.el
│   ├── casual-dired-variables.elc
│   ├── casual-dired-version.el
│   ├── casual-dired-version.elc
│   └── Makefile
├── Makefile
├── README.org
├── scripts
│   └── read-version.sh
└── tests
    ├── casual-dired-test-utils.el
    ├── casual-dired-test-utils.elc
    ├── Makefile
    ├── test-casual-dired.el
    ├── test-casual-dired-settings.el
    ├── test-casual-dired-sort-by.el
    ├── test-casual-dired-utils.el
    ├── test-casual-dired-variables.el
    └── test-casual-dired-version.el
```

If I use `:ensure t` instead, the package contains only the following

```
casual-dired-20240828.448/
├── casual-dired-autoloads.el
├── casual-dired.el
├── casual-dired.elc
├── casual-dired-pkg.el
├── casual-dired-settings.el
├── casual-dired-settings.elc
├── casual-dired-sort-by.el
├── casual-dired-sort-by.elc
├── casual-dired-utils.el
├── casual-dired-utils.elc
├── casual-dired-variables.el
├── casual-dired-variables.elc
├── casual-dired-version.el
└── casual-dired-version.elc
```

Any idea?

**Comment by kickingvegas** [comment]
@zeeros Thanks for sharing this. I’ve yet to start testing the Casual packages against Emacs 30 which includes a more recent version of `use-package` that supports the `:vc` key. From the looks of it, `:vc` seems to be over-aggressive in trying to byte-compile every file in the repo instead of constraining it to only the source directory specified in `:lisp-dir` (in this case the directory `lisp/`). The error seems to be triggered when `use-package` tries to byte-compile the files in the `tests/` directory, which IMHO it should *not* be doing.

That said, I do not regularly use `use-package` and my knowledge of its behavior is not deep. Perhaps others with more experience with it can provide guidance here.
